client/adjudicate.go

```diff
@@ -71,7 +71,14 @@ func (c *Channel) startWatching() (watcher.StatesPub, watcher.AdjudicatorSub, er
 		State:  currentTx.State.Clone(),
 		Sigs:   currentTx.Sigs,
 	}
-	statesPub, eventsSub, err := c.client.watcher.StartWatchingLedgerChannel(c.Ctx(), signedState)
+
+	statesPub, eventsSub, err := func() (watcher.StatesPub, watcher.AdjudicatorSub, error) {
+		if c.IsLedgerChannel() {
+			return c.client.watcher.StartWatchingLedgerChannel(c.Ctx(), signedState)
+		} else {
+			return c.client.watcher.StartWatchingSubChannel(c.Ctx(), c.parent.ID(), signedState)
+		}
```

Comment:
Using the interface made me think whether it wouldn't be better to just have one interface StartWatching(Context, Channel, Parent)? 🤔 

Anyways, applied your suggestion.

Reply:
I thought about it, but then what should be the value of parent when we start watching for ledger channel ?

We may not have avoid using `nil` or equivalent, because, this API could be accessed locally or over remote. In case of accessing over remote, the API call could involve one or two programming languages (client stub & server) and serialization for transporting over network.